File: Csharp/LongestSubStr.cs
using System;
using System.Collections.Generic;

namespace Csharp
{
    public class LongestSubStr
    {
        public LongestSubStr()
        {
            Console.WriteLine(LengthOfLongestSubstring("asdasdasd"));
        }

        public int LengthOfLongestSubstring(string s)
        {
            int n = s.Length;
            int i = 0, j = 0, ans = 0;
            HashSet<char> set = new HashSet<char>();
            while(i<n && j<n){
                if(!set.Contains(s[j])){
                    set.Add(s[j++]);
                    ans = Math.Max(ans,j-i);
                }else{
                    set.Remove(s[i++]);
                }
            }
            return ans;
        }
    }
}

File: Python3/LongestSubstr.py
import math
class LongestSubstr:
    def lengthOfLongestSubstring(self, s: str) -> int:
        i=0
        j=0
        ans=0
        n=len(s)
        mySet = set([])
        while i<n and j<n:
            if s[j] not in mySet:
                mySet.add(s[j])
                j+=1
                ans = max(ans, j-i)
            else:
                mySet.remove(s[i])
                i+=1
        return ans



test = LongestSubstr()
   
print(test.lengthOfLongestSubstring("abcabcbb"))